### PR TITLE
0.1.0-nullsafety.1

### PR DESCRIPTION
**Pull request by IgorKilipenko**
Migrate to ffi: ^1.0.0.
Add force lib prefix env (for windows platform).

### CODE REVIEW

File: lib/src/util.dart [hunk 1]
```diff
@@ -42,25 +42,25 @@ class Util {
...
     final res = call(() => writeFunc(ptr));
-    ffi.free(ptr);
+    ffi.calloc.free(ptr);
     return res;
   }
 
   static String? fromUtf8(ffi.Pointer<ffi.Int8> str) {
     if (str == ffi.nullptr) return null;
-    final length = ffi.Utf8.strlen(str.cast());
+    final length = ffi.Utf8Pointer(str.cast()).length; // ffi.Utf8.strlen(str.cast());
```
Comment (jpnurmi) [inline lib/src/util.dart:63]: could you clean up the comment?

**Review by jpnurmi — COMMENTED** [review]
Thanks a lot for the FFI changes! The `dylib.dart` changes don't seem to be related to the rest of the PR, though?

I would actually prefer to take the [dylib](https://pub.dev/packages/dylib) package into use here (later in a separate change). Any chance to contribute this change to https://github.com/jpnurmi/dylib.dart and also explain the motivation? :)

File: lib/src/util.dart (hunk 1 above)
Comment (IgorKilipenko) [reply lib/src/util.dart]: I use lib prefix for win libs too) and wanted to leave the option to force the prefix) I removed this

Comment (jpnurmi) [reply lib/src/util.dart]: Oh, ok. For a moment I thought the lib prefix was wrong on Windows. Anyway, I think we could make this doable with dylib.dart

**Review by jpnurmi — APPROVED** [review]
Thanks!

**Comment by jpnurmi** [comment]
https://pub.dev/packages/dart_serial_port/versions/0.1.0-nullsafety.1

**Pull request merged**